add .pre-commit-config.yaml config (refer to file for install guide l…

arswright opened the pull request:
* add .pre-commit-config.yaml config (refer to file for install guide link)
* add phpcs.xml rule config for review

Would like to run these configurations by you before getting into cleanup.

Thinking I'll branch to "arswright/cleanup", write a script that runs phpcs/cbf on one file at a time, commit once per file after manual cleanup, and issue a PR for every top-level directory as it is completed.

Will merge from upstream regularly to make things easier to merge (hopefully prevent any need to rebase).

Once you have a pre-commit+phpcs config you're happy with, please ping me and I'll start cleanup passes.

Commit:
add .pre-commit-config.yaml config (refer to file for install guide link)

add phpcs.xml rule config for review

leonstafford commented:
Thanks @arswright - just seeing this now and will give it a look

leonstafford commented on phpcs.xml:
```diff
@@ -0,0 +1,76 @@
+	<rule ref="Generic.WhiteSpace.ScopeIndent">
+		<properties>
+			<property name="indent" value="2" />
+			<property name="tabIndent" value="true" />
```
@arswright able to do spaces?

leonstafford commented:
@arswright giving this a run now and will see what rules I may want to disable for now

leonstafford commented:
@arswright could we start with this:

removed the check for short array syntax, as we still need to support PHP 5.3, at least for activation, if not full functionality at this point (would also like to add the PHP5.3 compatibility checker to this.

Also removed the WP checkers for now. Do you have a tip for quickly getting the WordPress sniffers added to my global environment? I've simply aliased phpcs to the full path of a clone of the project. 

```
<?xml version="1.0"?>
<ruleset name="temporary">
	<description>Temporary coding standard pending project maintainer review.</description>
	<arg name="colors" />
	<arg name="tab-width" value="2" />
	<!-- Custom line length validation -->
	<rule ref="Generic.Files.LineLength">
		<properties>
			<property name="lineLimit" value="80" />
			<property name="absoluteLineLimit" value="100" />
		</properties>
	</rule>
	<rule ref="Generic.WhiteSpace.ScopeIndent">
		<properties>
			<property name="indent" value="2" />
			<property name="tabIndent" value="false" />
		</properties>
	</rule>
  <rule ref="Generic.CodeAnalysis.ForLoopWithTestFunctionCall.NotAllowed" />
  <rule ref="Generic.Formatting.MultipleStatementAlignment.NotSameWarning" />
  <rule ref="Generic.Functions.OpeningFunctionBraceKernighanRitchie.BraceOnNewLine" />
  <rule ref="Generic.PHP.NoSilencedErrors.Discouraged" />
  <rule ref="Generic.Strings.UnnecessaryStringConcat.Found" />
  <rule ref="PSR2.Methods.MethodDeclaration.Underscore" />
  <rule ref="Squiz.PHP.CommentedOutCode.Found" />
  <rule ref="Squiz.PHP.DisallowMultipleAssignments.Found" />
  <rule ref="Squiz.PHP.EmbeddedPhp.ContentAfterOpen" />
  <rule ref="Squiz.PHP.EmbeddedPhp.ContentAfterEnd" />
  <rule ref="Squiz.PHP.EmbeddedPhp.ContentBeforeOpen" />
  <rule ref="Squiz.PHP.EmbeddedPhp.ContentBeforeEnd" />
  <rule ref="Squiz.PHP.NonExecutableCode.Unreachable" />
</ruleset>
```

ghost commented:
> Also removed the WP checkers for now.

I'd recommend keeping them if at all possible - I've got the more-annoying ones disabled but the deprecated function checks are really handy.

> Do you have a tip for quickly getting the WordPress sniffers added to my global environment?

IIRC I changed permissions on my phpcs standards directory (`/usr/share/php/PHP/CodeSniffer/src/Standards/`) to allow write for my uid and cloned the WP sniff repo into it

leonstafford commented:
ok, cool. let's keep them in. 

can you add the scripts to this PR and I'll merge it all in at once?

ghost commented:
OK - will do.

Commit:
Merge branch 'master' of https://github.com/leonstafford/wordpress-static-html-plugin into arswright/precommit_config

Commit:
updated configuration per issue discussion
URL: https://github.com/leonstafford/wordpress-static-html-plugin/pull/158

ghost commented:
This PR is going to be very noisy for a while...

If you want to unwatch it for now, I can ping you once I have something substantial completed - fair?

leonstafford commented:
done - muted the thread :smile_cat:

Commit:
add utility script and supporting files (todo: delete from branch at completion)

Commit:
Merge branch 'master' of https://github.com/leonstafford/wordpress-static-html-plugin into arswright/precommit_config

Commit:
code cleanup

Commit:
utility script bugfix

Commit:
code cleanup

Commit:
code cleanup

Commit:
adding file header comments

Commit:
adding file header comments

Commit:
prune file list (limit to plugin files, no phpcs/phpcbf for vendor files)

Commit:
additional rule exclusions to override wordpress standard's tab requirement

code cleanup

Commit:
add rule exclusions for fclose/fread/fwrite

code cleanup

Commit:
code cleanup

Commit:
code cleanup

Commit:
code cleanup

Commit:
code cleanup

Commit:
skip library/StaticHtmlOutput/MimeTypes.php - not likely to change anytime soon

Commit:
code cleanup

Commit:
code cleanup

Commit:
code cleanup

Commit:
code cleanup

Commit:
code cleanup

Commit:
code cleanup w/@leonstafford note RE: missing `break;`

Commit:
code cleanup w/comment @leonstafford RE: $this in $this

Commit:
code cleanup

Commit:
code cleanup

Commit:
code cleanup

Commit:
code cleanup

Commit:
skip UrlHelper

Commit:
code cleanup

Commit:
code cleanup

Commit:
skipping main plugin file

line length limit: assinine if you're using an IDE but nice-to-have
for those using vi/vim/emacs

... unfortunately, line length limit also breaks plugin data parser
(workaround..?)

Commit:
code quality for tests? nah, not today.

Commit:
remove helper script and supporting files to prep for rebase

leonstafford commented:
closing this PR as it's no longer linked to the repo. will create one from the branch instead
